### PR TITLE
New client takes LogLevel from config.

### PR DESCRIPTION
**Pull request by solaslin**
* Write the log level to "log-level" of new client .ini (Error/Terse/Versobe/Debug), originally it's always Debug
* Change "SolutionProgress" to "Idx/TotalSize" instead of "Idx/LastIdx", which is aligned with "ProblemProgress"
* When DID_NOT_SATISFY_ASSERT, print only the failed predicates to make it clean.

### CODE REVIEW

**Review by TonyYHsieh — COMMENTED** [review]
LGTM

**Review by aazz44ss — APPROVED** [review]
LGTM

File: Tensile/ClientWriter.py [hunk 1]
```diff
@@ -551,6 +551,12 @@ def dataInitParams(problemType):
             ('init-alpha', dataInitName(initAlpha)),
             ('init-beta',  dataInitName(initBeta))]
 
+def clientLogLevelName(num):
```
Comment (rosenrodt) [inline Tensile/ClientWriter.py:554]: Consider Enum class:

```python3
from enum import Enum
class LogLevel(Enum):
  Error = 0
  Terse = 1
  Verbose = 2
  Debug = 3

print(LogLevel.Debug.name)  # prints Debug
print(LogLevel.Debug.value) # prints 3
```

File: Tensile/Source/client/source/SolutionIterator.cpp [hunk 2]
```diff
@@ -141,7 +141,7 @@ namespace Tensile
         {
             m_reporter->report(ResultKey::SolutionIndex, m_currentSolutionIdx);
             m_reporter->report(ResultKey::SolutionProgress,
-                               concatenate(m_currentSolutionIdx, "/", m_lastSolutionIdx));
+                               concatenate(m_currentSolutionIdx, "/", m_lastSolutionIdx - m_firstSolutionIdx + 1));
```
Comment (sdquiring) [inline Tensile/Source/client/source/SolutionIterator.cpp:144]: Won't this end up printing e.g. 12/10 if m_firstSolutionIdx > 0?

Comment (solaslin) [reply Tensile/Source/client/source/SolutionIterator.cpp:144]: Looks like possible. 
Somehow I just want to make the ProblemProgress (Idx/Size) consistent with SolutionProgress(Idx/LastIdx).
So I guess it's also possible for ProblemProgress to print 12/10 ?
How about unify them both to Idx/LastIdx ?

File: Tensile/ClientWriter.py (hunk 1 above)
Comment (solaslin) [reply Tensile/ClientWriter.py]: Adopted. And also change the dataInitName to the same method for consistency

**Comment by sdquiring** [comment]
Looks good! There's an issue with the format check.  You can ignore the ubuntu16 ones the ubuntu18 one should be fixed.

**Pull request merged**